### ISSUE TITLE
Consider checking the type of a bad row before parsing directly with the right decoder

### ISSUE THREAD

**Issue by benjben** [snowplow-incubator/snowplow-badrows#35]
At the moment we use `.as[BadRow]` to parse a `Json` to its bad row case class.
This is based on [ADTs encoding and decoding](https://circe.github.io/circe/codecs/adt.html).
Given the number of classes and fields in the bad rows it might be more performant to first check the schema of the bad row (in `"schema"` field) and then use directly the right `Decoder`, e.g.
```scala
schema match {
  case Schemas.EnrichmentFailures.toSchemaUri =>
    json.as[EnrichmentFailures]
  ...
}
```

**Comment by chuwy** [comment]
Yup, that's a good idea. But I believe it is duplicated ticket: https://github.com/snowplow-incubator/snowplow-badrows/issues/28

**Comment by chuwy** [comment]
At the same time, I think it would be better to leave `Decoder[BadRow]` like it is now, because `.as[A]` decoding implies that structure is compatible with `A`, while what we really can decode it `SelfDescribingData[BadRow]`
